src/redux/waiting-room/saga.ts

```diff
@@ -31,55 +29,35 @@ export default function* waitingRoomSaga(
     updatedAt: new Date().getTime(),
   };
 
+  const challengeKey = `/challenges/${address}`;
+
   yield put(applicationActions.waitingRoomSuccess({ ...commonChallengeProps, stake }));
   const serializedChallenge = {
     ...commonChallengeProps,
     stake: stake.toString(),
   };
+  
+  yield firebase.database().ref(challengeKey).onDisconnect().remove();
```

Comment:
I _think_ you might need a `call` here for this to work reliably. Not 100% sure though.

Just to confirm, I'm guessing that this sends an instruction to firebase to remove the challenge on disconnect. Is that correct?

Reply:
@tomclose Correct, this is an instruction to Firebase to delete the entry at the reference on client disconnect (https://firebase.google.com/docs/reference/js/firebase.database.OnDisconnect#remove).

I don't believe a `call` is necessary, but it is a good idea to use `call` for the code to be testable.

As a note, one flaw with this approach is that, if a client disconnects and re-connects while in the Lobby, Firebase deletes the challenge but does not create a new challenge on re-connect. There is a listener that allows an elegant solution to this flaw (`firebase.database().ref(".info/connected")`), but I am not sure it is worth our time at the moment to worry about this edge case.